app/modules/zoo-api/zooapi.factory.js

```diff
@@ -4,10 +4,22 @@ require('./zooapi.module.js')
     .factory('zooAPI', zooAPI);
 
 var ApiClient = require('panoptes-client/lib/api-client');
-
+var OAuth = require('panoptes-client/lib/oauth')
 // @ngInject
-function zooAPI(zooAPIConfig) {
-
+function zooAPI(AnnotationsFactory, $location, $rootScope) {
+    ApiClient.beforeEveryRequest = function() {
+        return OAuth.checkBearerToken()
+            .then(function (token) {
+                console.log('Token refreshed: ', token);
+            })
+            .catch(function (error) {
+                console.log('Failed to refresh token: ', error);
+                alert('Your session is about to expire. Press "OK" to save your work and get a new session.');
+                AnnotationsFactory.updateCache();
+                OAuth.signIn($location.absUrl());
+                $rootScope.$broadcast('auth:loginChange');
```

Comment:
Sign in redirects the browser to zooniverse.org, to get a new token, so this line will never run. Since we're leaving shakespearesworld.org, and reloading it from scratch, do you need to broadcast an event here?

Reply:
Good point, we don't need that.